bee-tangle/src/metadata.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::flags::Flags;
+
+use bee_common::packable::{OptionError, Packable, Read, Write};
+use bee_ledger::types::ConflictReason;
+use bee_message::{milestone::MilestoneIndex, MessageId};
+
+use serde::Serialize;
+
+use std::{
+    cmp::Ordering,
+    time::{SystemTime, UNIX_EPOCH},
+};
+
+/// Metadata associated with a tangle message.
+#[derive(Copy, Clone, Default, Debug, Serialize)]
+pub struct MessageMetadata {
+    flags: Flags,
+    milestone_index: Option<MilestoneIndex>,
+    arrival_timestamp: u64,
+    solidification_timestamp: u64,
+    reference_timestamp: u64,
+    omrsi: Option<IndexId>,
+    ymrsi: Option<IndexId>,
+    conflict: ConflictReason,
+}
+
+impl MessageMetadata {
+    /// Create a new instance of a message's metadata.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        flags: Flags,
+        milestone_index: Option<MilestoneIndex>,
+        arrival_timestamp: u64,
+        solidification_timestamp: u64,
+        reference_timestamp: u64,
+        omrsi: Option<IndexId>,
+        ymrsi: Option<IndexId>,
+        conflict: ConflictReason,
+    ) -> Self {
+        Self {
+            flags,
+            milestone_index,
+            arrival_timestamp,
+            solidification_timestamp,
+            reference_timestamp,
+            omrsi,
+            ymrsi,
+            conflict,
+        }
+    }
+
+    /// Create metadata that corresponds to a just-arrived message using the current system time.
```

Comment:
Docs: 3rd person. Main reasons:
* consistency with the rest of the codebase
* consistency with the ecosystem or at least `std`

Reply:
I'm really not sure what you're talking about, you'll have to clarify.

Reply:
`Creates....`

Reply:
I think this is bike-shedding and a very weird hill to die on. It's not even like `std` has consistency on this, even.

https://doc.rust-lang.org/std/fs/fn.read.html

"Read" vs "Reads"